pkg/controller/configmap/configmap_controller.go

```diff
@@ -0,0 +1,133 @@
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("configmap-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource Group
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		CreateFunc: func(createEvent event.CreateEvent) bool {
+			return createEvent.Meta.GetName() == userCABundleConfigMapName && createEvent.Meta.GetNamespace() == names.ADDL_TRUST_BUNDLE_CONFIGMAP_NS
+
+		},
+		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
+			return deleteEvent.Meta.GetName() == userCABundleConfigMapName && deleteEvent.Meta.GetNamespace() == names.ADDL_TRUST_BUNDLE_CONFIGMAP_NS
+		},
+		UpdateFunc: func(updateEvent event.UpdateEvent) bool {
+			return updateEvent.MetaNew.GetName() == userCABundleConfigMapName && updateEvent.MetaNew.GetNamespace() == names.ADDL_TRUST_BUNDLE_CONFIGMAP_NS
+		},
+		GenericFunc: func(genericEvent event.GenericEvent) bool {
+			return genericEvent.Meta.GetName() == userCABundleConfigMapName && genericEvent.Meta.GetNamespace() == names.ADDL_TRUST_BUNDLE_CONFIGMAP_NS
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileGroup implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileConfigMap{}
+
+// ReconcileConfigMap reconciles a ConfigMap object
+type ReconcileConfigMap struct {
+	client            client.Client
+	scheme            *runtime.Scheme
+	metricsAggregator *metrics.AdoptionMetricsAggregator
+}
+
+// Reconcile reads that state of the cluster for a ConfigMap object and makes changes based the contained data
+func (r *ReconcileConfigMap) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ConfigMap")
+
+	// Fetch the ConfigMap openshift-config/user-ca-bundle
+	cfgMap := &corev1.ConfigMap{}
+	ns := names.ADDL_TRUST_BUNDLE_CONFIGMAP_NS
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: ns, Name: userCABundleConfigMapName}, cfgMap)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	certBundle, _, err := validation.TrustBundleConfigMap(cfgMap)
+	if err != nil {
+		// Reporting of failed validation is already funneled through CVO via ClusterOperatorDegraded. This is to explicity catch this condition
+		// and handle gracefully rather then causing stacktrace.
+		if strings.Contains(err.Error(), "failed to parse certificate PEM") {
```

Comment:
In my testing of this PR I was able to induce a different failure which didn't match against this test and never set the metric:

```
{"level":"error","ts":1639987467.7797878,"logger":"controller","msg":"Reconciler error","controller":"configmap-controller","name":"user-ca-bundle","namespace":"openshift-config","error":"failed parsing certificate data from ConfigMap \"user-ca-bundle\": failed to parse certificate: asn1: structure error: length too large","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.4.0/zapr.go:132\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.5/pkg/internal/controller/controller.go:246\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.5/pkg/internal/controller/controller.go:218\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.6.5/pkg/internal/controller/controller.go:197\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\tpkg/mod/k8s.io/apimachinery@v0.19.7/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\tpkg/mod/k8s.io/apimachinery@v0.19.7/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tpkg/mod/k8s.io/apimachinery@v0.19.7/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\tpkg/mod/k8s.io/apimachinery@v0.19.7/pkg/util/wait/wait.go:90"}
```

Maybe just `failed to parse certificate` is safe enough to match on.